build/Jenkinsfile.pr

```diff
@@ -237,336 +237,340 @@ pipeline {
-						/usr/local/bin/docker-compose --no-ansi run --rm start_dependencies
-						/usr/local/bin/docker-compose --no-ansi ps
-						cat ../tests/${LDAP_DATA}-data.ldif | /usr/local/bin/docker-compose --no-ansi exec -T openldap bash -c 'ldapadd -x -D "cn=admin,dc=mm,dc=test,dc=com" -w mostest';
-						"""
-					}
-				}
-
-				dir('src/github.com/mattermost/mattermost-server') {
-					ansiColor('xterm') {
-						sh """
-						echo "Creating databases"
-						/usr/local/bin/docker-compose --no-ansi -f build/docker-compose.yml  exec -T postgres sh -c 'exec echo "CREATE DATABASE migrated; CREATE DATABASE latest;" | exec psql -U mmuser mattermost_test'
-						echo "Importing postgres dump from version 5.0"
-						/usr/local/bin/docker-compose --no-ansi -f build/docker-compose.yml  exec -T postgres psql -U mmuser -d migrated < \$(pwd)/scripts/mattermost-postgresql-5.0.sql
-						"""
+		stage('Test') {
+			failFast true
+			parallel {
+				stage('Test-Mysql') {
+					environment {
+						GOPATH = "/go"
+						GOBIN = "/go/src/github.com/mattermost/mattermost-server/bin"
+						TEST_DATABASE_MYSQL_DSN = "mmuser:mostest@tcp(mysql:3306)/mattermost_test?charset=utf8mb4,utf8\u0026readTimeout=30s\u0026writeTimeout=30s"
+						TEST_DATABASE_POSTGRESQL_DSN = "postgres://mmuser:mostest@postgres:5432/mattermost_test?sslmode=disable&connect_timeout=10"
+						TEST_DATABASE_MYSQL_ROOT_PASSWD = "mostest"
+						MM_SQLSETTINGS_DRIVERNAME = "mysql"
+						MM_SQLSETTINGS_DATASOURCE = "mmuser:mostest@tcp(mysql:3306)/mattermost_test?charset=utf8mb4,utf8"
+						CI_INBUCKET_HOST = "inbucket"
+						CI_MINIO_HOST = "minio"
+						CI_INBUCKET_PORT = "10080"
+						CI_MINIO_PORT = "9000"
+						CI_INBUCKET_SMTP_PORT = "10025"
+						CI_LDAP_HOST = "openldap"
+						IS_CI = true
+						MM_EMAILSETTINGS_SMTPSERVER = "inbucket"
+						MM_EMAILSETTINGS_SMTPPORT = "10025"
+						MM_ELASTICSEARCHSETTINGS_CONNECTIONURL = "http://elasticsearch:9200"
+						LDAP_DATA = "test"
 					}
-				}
-
-				withDockerContainer(args: "-u root --privileged --net ${COMPOSE_PROJECT_NAME}_mm-test -v ${WORKSPACE}/src:/go/src/", image: 'mattermost/mattermost-build-server:oct-18-2019') {
-					ansiColor('xterm') {
-						sh """
-							cd /go/src/github.com/mattermost/mattermost-server
-							echo "Setting up config for db migration"
-							export MM_SQLSETTINGS_DATASOURCE=\"postgres://mmuser:mostest@postgres:5432/migrated?sslmode=disable&connect_timeout=10\"
-							export MM_SQLSETTINGS_DRIVERNAME=\"postgres\"
-							make ARGS="config get SqlSettings.DataSource" run-cli
-							echo "Running the migration"
-							make ARGS="version" run-cli
-
-							echo "Setting up config for fresh db setup"
-							export MM_SQLSETTINGS_DATASOURCE=\"postgres://mmuser:mostest@postgres:5432/latest?sslmode=disable&connect_timeout=10\"
-							make ARGS="config get SqlSettings.DataSource" run-cli
-
-							echo "Setting up fresh db"
-							make ARGS="version" run-cli
-						"""
-					}
-				}
 
+					steps {
```

Comment:
I think we can extract this to the stage "Test" as a common part that is not run in parallel